request_recorder.go

```diff
@@ -18,27 +19,42 @@ func newRequestRecorder() *requestRecorder {
 }
 
 type requestRecorder struct {
+	mtx               sync.RWMutex
 	acceptedRequests  []recordedRequest
 	unmatchedRequests []recordedRequest
 }
 
 func (r *requestRecorder) AcceptedRequests() []recordedRequest {
-	return r.acceptedRequests
+	r.mtx.RLock()
+	defer r.mtx.RUnlock()
+	res := make([]recordedRequest, len(r.acceptedRequests))
+	copy(res, r.acceptedRequests)
+	return res
```

Comment:
Can be extracted and reused for better readability. For example:
```go
return copyOf(r.acceptedRequests)
```
Where:
```go
func copyOf(orig []recordedRequest) []recordedRequest {
    res := make([]recordedRequest, len(orig))
    copy(res, orig)
    return res
}
```

Reply:
Refactored 👍